de.dlr.sc.virsat.model.extension.ps/src/de/dlr/sc/virsat/model/extension/ps/dnd/ProductStructureDragAndDropInheritanceCommandHelper.java

```diff
@@ -0,0 +1,165 @@
+	 * @param dragObjects The objects preferably SEIs which are dragged
+	 * @param dropSei the SEI to which the objects shall get dropped
+	 * @return A BeanSEI with corresponding to the drop target with correct name and discipline
+	 */
+	public IBeanStructuralElementInstance createBeanStructuralElementInstanceForDrop(VirSatTransactionalEditingDomain ed, Collection<Object> dragObjects, StructuralElementInstance dropSei) {
+		// When creating a new object and setting the inheritance link at the same time
+		// then there should be only one element in the list anyway, so get the first one.
+		Object dragObject = dragObjects.iterator().next();
+		
+		try {
+			if (dragObject instanceof StructuralElementInstance) {
+				// Identify the concept
+				VirSatResourceSet virSatResourceSet = ed.getResourceSet();
+				Repository currentRepository = virSatResourceSet.getRepository();
+				Concept concept = new ActiveConceptHelper(currentRepository).getConcept(Activator.getPluginId());
+				
+				// Create a bean for the given dropSEI
+				BeanStructuralElementInstanceFactory bsf = new BeanStructuralElementInstanceFactory();
+				IBeanStructuralElementInstance dropBeanSei = bsf.getInstanceFor(dropSei);
+				
+				// Use the created bean to decide which type of bean should be created for adding to the dropSEI
+				Class<? extends IBeanStructuralElementInstance> dropBeanSeiClass = dropBeanSei.getClass();
+				Class<? extends IBeanStructuralElementInstance> createBeanSeiClass = mapDropBeanToCreateBean.get(dropBeanSeiClass);
+				
+				// If there is a suggestion to it and if it was successful to identify the concept
+				// Then it it is possible to create a new BeanSei and to set it up correctly
+				if (createBeanSeiClass != null && concept != null) {
+					Constructor<? extends IBeanStructuralElementInstance> createBeanSeiConstructor = createBeanSeiClass.getDeclaredConstructor(Concept.class);
+					IBeanStructuralElementInstance createBeanSei = (IBeanStructuralElementInstance) createBeanSeiConstructor.newInstance(concept);
+					StructuralElementInstance createSei = createBeanSei.getStructuralElementInstance();
+					
+					// Now copy the name and the disciplines accordingly
+					StructuralElementInstance dragSei = (StructuralElementInstance) dragObject;
+					Discipline discipline = dropSei.getAssignedDiscipline(); 
+					createSei.setAssignedDiscipline(discipline);
+					String name = dragSei.getName(); 
+					createSei.setName(name);
+					
+					// Now fix naming clashes
+					DVLMCopiedNameHelper dcnh = new DVLMCopiedNameHelper();
+					dcnh.updateCopiedNames(dropSei.getChildren(), Collections.singleton(createSei));
+	
+					return createBeanSei;
+				}
+			}
+		} catch (CoreException | NoSuchMethodException | SecurityException | InstantiationException | IllegalAccessException | IllegalArgumentException | InvocationTargetException e) {
+			Activator.getDefault().getLog().log(new Status(
+					Status.INFO,

```

Comment:
Is this really just an info and not an error?

Reply:
Well a WARNING could be ok, ERROR I think should indicate a problem that may create havoc when further executing the application. WARNING is more like something went unexpectedly wrong but is regarded as no critical problem.